src/main/java/mate/academy/spring/dao/impl/RoleDaoImpl.java

```diff
@@ -0,0 +1,25 @@
+package mate.academy.spring.dao.impl;
+
+import mate.academy.spring.dao.AbstractDao;
+import mate.academy.spring.dao.RoleDao;
+import mate.academy.spring.exception.DataProcessingException;
+import mate.academy.spring.model.Role;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class RoleDaoImpl extends AbstractDao<Role> implements RoleDao {
+    public RoleDaoImpl(SessionFactory factory) {
+        super(factory, Role.class);
+    }
+
+    @Override
+    public Role getByName(String roleName) {
```

Comment:
```suggestion
    public Role getByName(Role.RoleName roleName) {
```
![image](https://user-images.githubusercontent.com/117104543/222989640-05b41432-2bad-4f62-95db-436cf1192c03.png)

Reply:
I do this logic in DAO layer